### ISSUE TITLE
Setting up objects prior to running a SCENARIO

### ISSUE THREAD

**Issue by devhandle** [catchorg/Catch2#1406]
QUESTION: 

I'm writing tests using the format below. The problem is that SCENARIO is called every time a new _GIVEN_ test is executed. This makes declaring variables inside SCENARIO troublesome - I have a logger class which fails because I'm instantiating it multiple times. 

I've moved all vars to an anonymous namespace which works but looks a bit messy. Is this correct way to do it or is there somewhere I can instantiate/setup objects before my test runs. 

Appreciate any help.

```
#include "single_include/catch2/catch.hpp"

SCENARIO ("My Scenario","[MyClass]")
{
    MyClass myClass;

    GIVEN ("...")
    {
        WHEN ("..")
        {
            THEN ("...")
            {
                REQUIRE(condition);
            }
        }
    }
    GIVEN ("...")
    {
        WHEN ("..")
        {
            THEN ("...")
            {
                REQUIRE(condition);
            }
        }
    }
}
```

**Comment by horenmar** [comment]
If I understand correctly what you want to do, you want the [Event Listener interface](https://github.com/catchorg/Catch2/blob/master/docs/event-listeners.md#event-listeners).

**Comment by devhandle** [comment]
I'm trying to write a BDD style test using [this].(https://github.com/catchorg/Catch2/blob/master/examples/120-Bdd-ScenarioGivenWhenThen.cpp) style. The tests (each _WHEN_) should run one after the other in sequence. I've copied this exact format  as the example but it seems _GIVEN_ is called multiple times meaning my variable is redeclared for each WHEN condition and loses it's state. I've added logging which confirms this. 

Is it possible to to run each _WHEN_ in sequence and keep variable state between each one?
